File: tasks/2026/round2_pp/robot-pp.py
#!/usr/bin/env python3
# @check-accepted: examples NMsmall RD no-limits


def main():
    N, M, K = map(int, input().split())
    P = input().strip()

    free_cells = N * M - 1
    vis = [[False] * M for _ in range(N)]
    vis[0][0] = True

    x, y = 0, 0

    for i, c in enumerate(P):
        if c == "R":
            y += 1
        elif c == "L":
            y -= 1
        elif c == "U":
            x -= 1
        elif c == "D":
            x += 1

        if not vis[x][y]:
            vis[x][y] = True
            free_cells -= 1

        if free_cells == K - 1:
            print(i)
            return

    print(len(P))


if __name__ == "__main__":
    main()

File: tasks/2026/round2_pp/robot-pp.cpp
// @check-accepted: examples NMsmall RD no-limits

#include <fstream>
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N, M, K;
    cin >> N >> M >> K;
    string P;
    cin >> P;
    int free_cells = N * M - 1;
    vector<vector<bool> > vis(N, vector<bool>(M, false));
    vis[0][0] = true;
    int x = 0, y = 0;
    for (int i = 0; i < P.size(); i++) {
        auto c = P[i];
        if (c == 'R') {
            y++;
        } else if (c == 'L') {
            y--;
        } else if (c == 'U') {
            x--;
        } else if (c == 'D') {
            x++;
        }
        if (!vis[x][y]) {
            vis[x][y] = true;
            free_cells--;
        }
        if (free_cells == K - 1) {
            cout << i << endl;
            return 0;
        }
    }
    cout << P.size() << endl;
}
